shared/features/invoice/handlers/fetch-font-info.ts

```diff
@@ -0,0 +1,125 @@
+  ["hin"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Devanagari&display=swap",
+  ["tha"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Thai&display=swap",
+  ["arb"]: "https://fonts.googleapis.com/css2?family=Noto+Kufi+Arabic&display=swap",
+  ["ell"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["chn"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+SC&display=swap",
+  ["rus"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["jpn"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+JP&display=swap",
+  ["grc"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["kor"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+KR&display=swap",
+  ["sau"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Arabic&display=swap",
+  ["irn"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Arabic&display=swap",
+  ["egy"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Arabic&display=swap",
+  ["ind"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Devanagari&display=swap",
+  ["isr"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Hebrew&display=swap",
+  ["arm"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Armenian&display=swap",
+  ["geo"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Georgian&display=swap",
+  ["ukr"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["srb"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["blr"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["bgr"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["kaz"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+  ["mmr"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Myanmar&display=swap",
+  ["khm"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Khmer&display=swap",
+  ["lao"]: "https://fonts.googleapis.com/css2?family=Noto+Sans+Lao&display=swap",
+  ["vnm"]: "https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap",
+};
+
+const latinScriptCountries = [
+  "arg", // Argentina
+  "aus", // Australia
+  "aut", // Austria
+  "bel", // Belgium
+  "bra", // Brazil
+  "can", // Canada
+  "che", // Switzerland
+  "chl", // Chile
+  "col", // Colombia
+  "cri", // Costa Rica
+  "cze", // Czech Republic
+  "deu", // Germany
+  "dnk", // Denmark
+  "esp", // Spain
+  "fin", // Finland
+  "fra", // France
+  "gbr", // United Kingdom
+  "grc", // Greece
+  "hun", // Hungary
+  "irl", // Ireland
+  "ita", // Italy
+  "mex", // Mexico
+  "nld", // Netherlands
+  "nor", // Norway
+  "nzl", // New Zealand
+  "pol", // Poland
+  "prt", // Portugal
+  "rou", // Romania
+  "swe", // Sweden
+  "usa", // United States
+  "ury", // Uruguay
+];
+
+export async function detectLanguageAndGetFontDynamically(text: string): Promise<FontInfo> {
+  const languageCode = franc(text);
+  console.info("Detected language :", { languageCode, text });
```

Comment:
Need logs ?

Reply:
Yes if we have a font error on server we need to find what (exotic) language it is